SDWebImage/SDWebImageManager.m

```diff
@@ -209,14 +209,19 @@ - (void)diskImageExistsForURL:(nullable NSURL *)url
                     if ([self.delegate respondsToSelector:@selector(imageManager:shouldBlockFailedURL:withError:)]) {
                         shouldBlockFailedURL = [self.delegate imageManager:self shouldBlockFailedURL:url withError:error];
                     } else {
-                        shouldBlockFailedURL = (   error.code != NSURLErrorNotConnectedToInternet
-                                                && error.code != NSURLErrorCancelled
-                                                && error.code != NSURLErrorTimedOut
-                                                && error.code != NSURLErrorInternationalRoamingOff
-                                                && error.code != NSURLErrorDataNotAllowed
-                                                && error.code != NSURLErrorCannotFindHost
-                                                && error.code != NSURLErrorCannotConnectToHost
-                                                && error.code != NSURLErrorNetworkConnectionLost);
+                        // Filter the error domain and check error codes
+                        if ([error.domain isEqualToString:NSURLErrorDomain]) {
+                            shouldBlockFailedURL = (   error.code != NSURLErrorNotConnectedToInternet
+                                                    && error.code != NSURLErrorCancelled
+                                                    && error.code != NSURLErrorTimedOut
+                                                    && error.code != NSURLErrorInternationalRoamingOff
+                                                    && error.code != NSURLErrorDataNotAllowed
+                                                    && error.code != NSURLErrorCannotFindHost
+                                                    && error.code != NSURLErrorCannotConnectToHost
+                                                    && error.code != NSURLErrorNetworkConnectionLost);
```

Comment:
Why merged to `master`? this break the usage.

Reply:
Emm, break what usage ?

Reply:
Previously, if domain is not `NSURLErrorDomain`, it block the url. Now it's not block. For example `SDWebImageErrorDomain`. I remember I revert this PR before? this is not the correct fix, we need to find another way. I'll revert this.

Reply:
Did your project using the code like this ?

It's not a correct usage of NSError, developer bug. The framework should not try to assume this. A error code must bind their domain to show the meannings.

If you think this is really important. Revert this only in the 4.x branch. Master branch is for the 5.0.0 release now. I've already checked out one and you can create MR here.